src/Persistence/AbstractQuery.php

```diff
@@ -284,8 +284,6 @@ public function execute()
 
 //          $this->model->hook(self::HOOK_BEFORE_EXECUTE, [$this]);
 
-            $this->hookOnModel('HOOK_BEFORE_' . $this->getMode(), [$this]);
```

Comment:
HOOK_BEFORE_EXECUTE is commented, but it seems this hook should be outer based on HOOK_AFTER_EXECUTE...

Reply:
Question is to keep `HOOK_BEFORE_INSERT`, `HOOK_BEFORE_UPDATE` or replace all with single `HOOK_BEFORE_EXECUTE`

Reply:
I would say no strongly - how would you otherwise hook only for update for example?

Reply:
You can distinguish using `$query->getMode()` in the closure

Reply:
I analysed `HOOK_(BEFORE)_(delete|update|insert)` matches now:

- `getMode()` in hook itself is ok for me, but user usually needs to hook for one type of operation only (for ex. I use it for Doctrine, but it can be easily replaced), like -2/10 helpful
- you probably want it because:
![image](https://user-images.githubusercontent.com/2228672/90612356-37159b00-e208-11ea-9bb9-66d80f5c7074.png)
right? would one hook for these actually help? I do not think so, <-3, 2>/10 helpful
- 3x in Model:
![image](https://user-images.githubusercontent.com/2228672/90612549-7c39cd00-e208-11ea-88bf-346cfe2d24ff.png)
no sure how you want to solve this then... <-8,2>/10 helpful

Reply:
Thanks for analysis. Will stick to the separate hooks then, no problem

Reply:
yes, checking the result is probably the strongest argument + https://github.com/atk4/core/pull/246 (as with single hook the probability the hook is already executing is much higher)

Reply:
We will keep hook constants defined in Persistence as now.